docs/crs_compatibility.rst

```diff
@@ -0,0 +1,267 @@
+    import fiona
+
+    with fiona.open(...) as fds:
+        proj_crs = CRS.from_wkt(fds.crs_wkt)
+
+
+Converting from `pyproj.crs.CRS` for `fiona`
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. warning:: WKT2 is only supported in GDAL 3+
+
+If you want to be compatible across GDAL versions, you can do::
+
+    from distutils.version import LooseVersion
+
+    import fiona
+
+    proj_crs = CRS.from_epsg(4326)
+
+    if LooseVersion(fiona.__gdal_version__) < LooseVersion("3.0.0"):
+        fio_crs = proj_crs.to_wkt(WktVersion.WKT1_GDAL)
+    else:
+        # GDAL 3+ can use WKT2
+        fio_crs = dc_crs.to_wkt()
+
+    # with fiona.open(..., "w", crs_wkt=fio_crs) as fds:
+    #     ...
+
+
+geopandas
+---------
+
+https://github.com/geopandas/geopandas
+
+Also see the `geopandas guide for upgrading to use pyproj CRS class <https://geopandas.readthedocs.io/en/latest/projections.html#upgrading-to-geopandas-0-7-with-pyproj-2-2-and-proj-6>`__
+
+Preparing `pyproj.crs.CRS` for `geopandas`
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: python
+
+    import fiona
+    import geopandas
+    from pyproj.crs import CRS
+    from pyproj.enums import WktVersion
+
+    proj_crs = CRS.from_epsg(4326)
+
+    if LooseVersion(geopandas.__version__) >= LooseVersion("0.7.0"):
+        # geopandas uses pyproj.crs.CRS
+        geo_crs = proj_crs
+    elif LooseVersion(geopandas.__version__) >= LooseVersion("0.6.0"):
+        # this version of geopandas uses always_xy=True so WKT version is safe
+        if LooseVersion(fiona.__gdal_version__) < LooseVersion("3.0.0"):
+            geo_crs = proj_crs.to_wkt(WktVersion.WKT1_GDAL)
+        else:
+            # GDAL 3+ can use WKT2
+            geo_crs = dc_crs.to_wkt()
+    else:
+        geo_crs = dc_crs.to_proj4()
+
+
+`geopandas` to `pyproj.crs.CRS`
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+:meth:`pyproj.crs.CRS.from_user_input` can handle anything across the `geopandas`
+versions. The only gotcha would be if it is `None`.
+
+
+.. code-block:: python
+
+    import geopandas
+    from pyproj.crs import CRS
+
+    gdf = geopandas.read_file(...)
+    proj_crs = CRS.from_user_input(gdf.crs)
+
+
+cartopy
+-------
+
+https://github.com/SciTools/cartopy
+
+This may change in the future:
+`Port to use pyproj v2 <https://github.com/SciTools/cartopy/issues/1477>`__
+
+Preparing `pyproj.crs.CRS` for `cartopy.crs.CRS`
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: python
+
+    from cartopy.crs import CRS as cCRS, Globe
+    from pyproj.crs import CRS
+
+    proj_crs = CRS.from_epsg(4326)
+    globe = Globe(
+        ellipse=None,
+        semimajor_axis=proj_crs.ellipsoid.semi_major_metre,
+        semiminor_axis=proj_crs.ellipsoid.semi_minor_metre,
+        inverse_flattening=proj_crs.ellipsoid.inverse_flattening,
+    )
+    proj_dict = proj_crs.to_dict()
+    proj_dict["pm"] = proj_crs.prime_meridian.longitude
+    cart_crs = cCRS(proj_dict, globe=globe)
```

Comment:
Personally I think doing `import cartopy.crs` and then `cartopy.crs.CRS(proj_dcit,globe=globe)` is clearer to read than something as `cCRS` (which is a rather "obscure" abbreviation ;))

Reply:
That is a good point. I changed it to be `import cartopy.crs as ccrs` as that is consistent with what is in their docs.